PyStemmusScope/stemmus_scope.py

```diff
@@ -12,32 +13,95 @@
+    #TODO add docstring
+
+    #TODO add documentation links in msg below
+    if model_src_path.is_file():
+        msg = ("The model executable file can be used on a Unix system "
+            "where MCR is installed, see the documentaion.")
+        logger.info("%s", msg)
+        return True
+    if model_src_path.is_dir():
+        return False
+    msg = (
+        "Provide a valid path to an executable file or "
+        "to a directory containing model source codes, "
+        "see the documentaion.")
+    raise ValueError(msg)
+
+
+def _check_sub_process(sub_process: str):
+    if sub_process not in {"Octave" , "Matlab"}:
+        msg = (
+            "Set `sub_process` as Octave or Matlab to run the model using source codes."
+            "Otherwise set `model_src_path` to the model executable file, "
+            "see the documentaion.")
+        raise ValueError(msg)
+
+
+def _run_sub_process(args: list, cwd):
+    # pylint: disable=consider-using-with
+    result = subprocess.Popen(
+        args, cwd=cwd,
+        stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+        shell=True,
+    )
+    exit_code = result.wait()
+    stdout, stderr = result.communicate()
+    #TODO handle stderr properly
+    # when using octave, exit_code might be 139
+    # see issue STEMMUS_SCOPE_Processing/issues/46
+    if exit_code not in [0, 139]:
+        raise subprocess.CalledProcessError(
+            returncode=exit_code, cmd=args, stderr=stderr, output=stdout
+        )
+    if exit_code == 139:
+        logger.warning(stderr)
+
+    # TODO return log info line by line!
+    logger.info(stdout)
+    return stdout.decode('utf-8')
+
 
 class StemmusScope():
     """PyStemmusScope wrapper around Stemmus_Scope model.
     see https://gmd.copernicus.org/articles/14/1379/2021/
 
-    It sets the model with a configuration file and executable file.
-    It also prepares forcing and soil data for model run.
+    It sets the model with a configuration file and executable file. It also
+    prepares forcing and soil data for model run.
 
     Args:
         config_file(str): path to Stemmus_Scope configuration file. An example
         config_file can be found in tests/test_data in `STEMMUS_SCOPE_Processing
         repository <https://github.com/EcoExtreML/STEMMUS_SCOPE_Processing>`_
-        exe_file(str): path to Stemmus_Scope executable file.
+        model_src_path(str): path to Stemmus_Scope executable file or to a
+        directory containing model source codes.
+        sub_process(str, optional): use `Matlab` or `Octave`. It is optional if
+        `model_src_path` is a path to Stemmus_Scope executable file.
 
     Example:
         See notebooks/run_model_in_notebook.ipynb in `STEMMUS_SCOPE_Processing
         repository <https://github.com/EcoExtreML/STEMMUS_SCOPE_Processing>`_
     """
 
-    def __init__(self, config_file: str, exe_file: str):
+    def __init__(self, config_file: str, model_src_path: str, sub_process: str = None):
```

Comment:
I'm not sure if `sub_process` is the right name here. I would think [`interpreter`](https://en.wikipedia.org/wiki/Interpreter_(computing)) is more accurate.

Reply:
I like the suggestion. Replaced.